File: app/src/main/java/com/tomclaw/vika/di/AppComponent.java
package com.tomclaw.vika.di;

import com.tomclaw.vika.main.auth.AuthActivity;
import com.tomclaw.vika.main.home.MainActivity;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by Igor on 10.11.2017.
 */
@Singleton
@Component(modules = {AppModule.class, UserModule.class, ApiModule.class})
public interface AppComponent {

    void inject(MainActivity activity);

    void inject(AuthActivity activity);

}

File: app/src/main/java/com/tomclaw/vika/di/AppComponent.kt
package com.tomclaw.vika.di

import com.tomclaw.vika.screen.chat.di.ChatComponent
import com.tomclaw.vika.screen.chat.di.ChatModule
import com.tomclaw.vika.screen.home.di.HomeComponent
import com.tomclaw.vika.screen.home.di.HomeModule
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class])
interface AppComponent {

    fun homeComponent(module: HomeModule): HomeComponent

    fun chatComponent(module: ChatModule): ChatComponent

}
